shortfin/python/shortfin_apps/sd/components/manager.py

```diff
@@ -34,7 +34,7 @@ def __init__(self, device="local-task", device_ids=None):
                         raise ValueError(f"Device id {did} could not be parsed.")
             else:
                 selected = available
-            sb = sf.amdgpu.SystemBuilder(amdgpu_visible_devices=";".join(selected))
+            sb = sf.SystemBuilder(system_type="amdgpu")
```

Comment:
You can just create the SystemBuilder once:

```
sb = sf.SystemBuilder(system_type="amdgpu")
sb.visible_devices = get_selected_devices(sb)
self.ls = sb.create_system()
```

(i.e. you can set properties between construction of the SystemBuilder and create_system())

Note that the property takes a normal list of strings so no need to join.

It also looks like you dropped the visible_devices thing in this patch.

Reply:
yeah, that was for making sure the specified device IDs actually matched an available device. I don't really like it, so I'll pass in the raw args.